mlir/lib/Mitigation/Transforms/MitigationMethods/Zne.cpp

```diff
@@ -412,7 +411,7 @@ FlatSymbolRefAttr ZneLowering::getOrInsertFnWithoutMeasurements(Location loc,
 
     quantum::DeallocOp localDealloc = *fnWithoutMeasurementsOp.getOps<quantum::DeallocOp>().begin();
     rewriter.eraseOp(localDealloc);
-    quantum::removeQuantumMeasurements(fnWithoutMeasurementsOp, rewriter);
+    quantum::replaceQuantumMeasurements(fnWithoutMeasurementsOp, rewriter);
```

Comment:
Let's replace this in the grad lowering as well since the `removeQuantumMeasurements` turns out to be unsafe.

Reply:
Yes I will do it in a separate PR.